### DISCUSSION TITLE
CPU usage higher w/ Audiogridder than w/o

### DISCUSSION THREAD

**Discussion by UncoveredPitch** [apohl79/audiogridder#428] [Help] [answered]
I've connected two pretty old Macs but somehow I remember the betas that I tried last year working better. The problem is that, unless I set the system to something like +30 buffers(which is unusable for any parameter tweaking), the CPU usage on the main computer is much higher(5-6x) than when not using Audiogridder. It starts out ok but then spikes get progressively worse. 

Networking seems quite stable and all the plug-ins are being recognised. CPU usage on the server is how it should be. What am I missing here? Short video of the problem attached. 

https://user-images.githubusercontent.com/68796652/110713309-9e715780-81f9-11eb-9de1-0e80cf129a23.mp4

**Comment by apohl79** [comment]
You can check the server processing time in the stats window of the server. If it is reasonable low, it is your network, as the plugin mon is just showing how long it takes for the audio data to go and come back from the server.

Regarding cpu spikes, i don't understand your monitoring video and what you are measuring and don't know where the spikes are. But the latency in the plugin mon looks extremely high to me.

**Reply by UncoveredPitch** [reply]
Thanks a lot for your reply. I don't know what reasonable processing times are as I'm new to this. I've attached a screenshot of the stats window you mentioned. A 10x ping test in Network Utility gave a spread of 0.35-0.56ms. The connection is over Gigabit Ethernet and both Macs confirm this speed. 

As far as I understand it, the CPU usage in the DAW on each track should be near zero, and the calculations made on the server. If you look at my video again, this is clearly not the case. The values spike up to 3.5% which is more than any plug-in would use locally in Reaper. The CPU usage on the server is as expected. So it looks like the problem lies on the DAW/plug-in side. 

Also, I've tried many combinations of buffer settings but nothing works properly. If I set the buffer in Audiogridder to 30 blocks the spikes are almost gone but then the latency makes editing impossible. As you said you test in Reaper, I was hoping you would have seen the problem before and had a "fix" for it. Thanks!

<img width="262" alt="AudiogridderServerStats" src="https://user-images.githubusercontent.com/68796652/111009644-21c2b280-838c-11eb-9264-e49dfec7c474.png">

**Reply by apohl79** [reply]
I think daw's are assuming CPU usage by looking at the processing latency of a plugin. The longer it takes, the higher the "pressure" to fill the I/O buffers. The AG plugin is mostly waiting for data and does no audio computations. That waiting for data is probably causing the daw to show more CPU usage.

Try bypassing plugins in AG and see how that affects your figures.

**Reply by apohl79** [reply]
One additional question:

> I was hoping you would have seen the problem

What is the actual problem you have (except for the CPU meter in reaper)?

**Reply by UncoveredPitch** [reply]
The problem is that the system isn't stable. Yes, the CPU meter readings in Reaper move much more than in the Activity Monitor, but the longer the system runs, the more often the RT CPU spikes until there are dropouts. Like in the video I posted. 
If I open one of the UIs, the dropouts occur straight away. Also, the latency goes up the longer the system runs. 

I tried many combinations of buffer sizes in Reaper and Audiogridder. Nothing works reliably so maybe my combination of Macs doesn't work with Audiogridder. I think I will just have to give up and go back to freezing and using fewer plugins. I should add that Reamote is even worse, so it may not be a problem with your software.

**Comment by UncoveredPitch** [comment, marked as answer]
Ok despite all the problems I've had it now miraculously works! I kept on messing with my network settings and cables, re-started a few times and did some other desperate things and now everything looks like it did in the travel brochure. :)

What I can't figure out is why wi-fi needs to be on on the server machine for both screen sharing and Audiogridder to work. It seems unrelated to the task at hand, and this is probably why it took me so long to find it. 

Thanks to the dev for making a few suggestions that encouraged me to look at the network settings rather than endlessly fiddling with buffer sizes both in Reaper and Audiogridder. In fact, the +1 block buffer setting works best right now. Now of course the next round of testing with maxing out my server begins...

**Reply by YodAmygdala** [reply]
I'm having the same problem now. Any chance you could share your network settings after all these years? I have no idea what to try to tweak.

**Reply by UncoveredPitch** [reply]
> I'm having the same problem now. Any chance you could share your network settings after all these years? I have no idea what to try to tweak.

I would love to help you but:

1. Both the Macs I connected via Audiogridder have had logic board failures and are no longer serviceable. I now only have one Mac which so far hasn't needed any "support". 

2. Reaper has had an internal overhaul that drastically reduces CPU usage on busses and also on tracks that aren't playing. So even on my old system I could just about get by without Audiogridder. 

I think what I meant with my reply is that the solution in my case revolved around "mindlessly" changing network settings on both Macs, un-plugging cables and re-starting. What seemed to help the most was switching on wi-fi on the server. I couldn't really explain why it worked then and I certainly can't now—sorry and good luck!

**Comment by Volkhard** [comment]
I also tried to use Audiogridder as external Plugin Server (I use a DigiGrid M on a Waves Soundgrid Platform). Starting AudioGridder on my Master PC CPU-load heavily increases after insertion of the plugin:

![image](https://user-images.githubusercontent.com/81081592/111914709-6973c900-8a73-11eb-830b-5570faf4af3a.png)

Do you have an idea how to cope this problem?

**Reply by UncoveredPitch** [reply]
I'm sure you would rather have the dev's answer but do feel free to have look at my reply in MistaMin's ProTools thread on here about keeping latencies as short as possible. In my limited experience with networking and Audiogridder the only thing I can say with any amount of certainty is that transmission speed is king!

**Reply by apohl79** [reply]
Also, try to increase the IO buffer in your DAW and AG.

**Comment by bsjdsp** [comment]
When using audio gridder does the plugins feel like native plugins or Dan we tell it’s streaming and doesn’t feel like smooth native plugins

**Comment by CaBuffalo** [comment]
The answer is native. You can listen to it yourself.

On Tue, Mar 23, 2021 at 12:01 PM bsjdsp ***@***.***> wrote:

> When using audio gridder does the plugins feel like native plugins or Dan
> we tell it’s streaming and doesn’t feel like smooth native plugins
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/428#discussioncomment-519430>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6BRMT6I3PKSM7RGLCLTFDQPHANCNFSM4Y7D24GA>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Reply by UncoveredPitch** [reply]
I believe the poster only has one computer to test right now and wants to get an impression of how the system works with two.

In order to make the experience as native as possible, I've had good results with using one AudioGridder plug-in for each native plug-in I want to be calculated on the server. Meaning not having chains of plug-ins inside one AudioGridder plug-in. This way multiple UIs can be open and edited simultaneously, like you can in a native session. 

If you have a DAW like Reaper that can rename plug-ins, even better. The delay in tweaking a parameter and hearing the result depends on your network speed and how low a buffer you can run. If your system is on the slow side then it will feel different from native, if it's well set up then it's very close. Right now mine is very close. Fingers crossed...
